dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -16,6 +16,33 @@ properties:
   pinctrl-names:
     required: true
 
+  st,encoder-mode:
+    type: int
+    description: |
+      Set encoder mode.
+      The following modes are supported:
+      0x1: Encoder mode 1
+      0x2: Encoder mode 2
+      0x3: Encoder mode 3
+      Some MCUs supports additional modes:
+      0x10002: Encoder mode: Clock plus direction, x2 mode
+      0x10003: Encoder mode: Clock plus direction, x1 mode
+      0x10004: Encoder mode: Directional Clock, x2 mode
+      0x10005: Encoder mode: Directional Clock, x1 mode
+      0x10006: Quadrature encoder mode: x1 mode, counting on tim_ti1fp1
+      0x10007: Quadrature encoder mode: x1 mode, counting on tim_ti2fp2
+    default: 0x1
```

Comment:
This needs a justification. Is it a register reset value? Please document in the description why the default value was chosen

Reply:
Thanks. I added default value documentation.

Reply:
Sorry, but one last comment on this part:
This will render as follows in doc: https://docs.zephyrproject.org/latest/build/dts/api/bindings/sensor/st%2Cstm32-qdec.html#dtbinding-st-stm32-qdec
As you can see, the description part is displayed, but not the default. So I, d' suggest to do something like:
```
    description: |
      Set encoder mode.
      The following modes are supported:
      0x1: Encoder mode 1 (Default)
      0x2: Encoder mode 2
      0x3: Encoder mode 3
 [...]
```